src/ansari/presenters/whatsapp_presenter.py

```diff
@@ -70,7 +68,7 @@ async def process_and_reply_to_whatsapp_sender(
     async def extract_relevant_whatsapp_message_details(
         self,
         body: Dict[str, Any],
-    ) -> Optional[Tuple[str, str, str]]:
+    ) -> Tuple[str, str, str] | str | None:
```

Comment:
This is only supported in Python 3.11 and up. For max flexibility and to allow people to use older versions of python, we should use the Union[] class.